scs/scsobject.h

```diff
@@ -563,10 +568,16 @@ static int SCS_init(SCS *self, PyObject *args, PyObject *kwargs) {
   self->sol->s = (scs_float *)scs_calloc(self->m, sizeof(scs_float));
 
   /* release the GIL */
+#ifdef Py_GIL_DISABLED
```

Comment:
Does the GIL protect against the race you're worried about here? If not then you should lock on the GIL-enabled build too. It'd be nice to have an explicit test case for this fix if you don't already.

Reply:
The general idea here was to lock before `Py_BEGIN_ALLOW_THREADS`.

Reply:
Oh oops, I completely missed that this is happening inside `SCS_init`. In that case locking probably isn't necessary, unless it's somehow possible for two threads to simultaneously see the same `SCS` object before `SCS_init` returns an initialized object.